gitpod-io/gitpod
# Add new payment validation flow in Dashboard

## Description

This adds support for a new payment validation flow when subscribing that supports more advanced verification, such as 3DS. 

It's behind the `paymentVerificationFlow` feature flag, which is off in prod/dedicated, but enabled for non-prod.

| Add Payment | Test 3DS Flow | Error Handling |
|--------|--------|--------|
| <img width="554" alt="image" src="https://user-images.githubusercontent.com/367275/234121147-b3424b24-a1b3-480b-9fa2-277e2b7a18ce.png"> | <img width="595" alt="image" src="https://user-images.githubusercontent.com/367275/234121332-04556dd6-39ef-4d26-9a69-a9df97fe2111.png"> | <img width="577" alt="image" src="https://user-images.githubusercontent.com/367275/234121390-e5a93a2b-7205-494e-8269-bc5749ed6aaa.png"> | 


## How to test

Visit the preview environment. Create a new org, and attempt to upgrade to a paid plan w/ credit card info.

You can use different Stripe [test credit cards](https://stripe.com/docs/testing#authentication-and-setup) that should force a 3DS flow. We should test various types, making sure we handle cases that require it, or don't, or ones that should error out as well.

## Related Issue(s)

Fixes WEB-196

## Documentation


#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - bmh-valida3889bb903d</li>
	<li><b>🔗 URL</b> - <a href="https://bmh-valida3889bb903d.preview.gitpod-dev.com/workspaces" target="_blank">bmh-valida3889bb903d.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Build Options:

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

</details>

<details>
<summary>Installer Options</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

/hold
